src/docfx/build/xref/XrefResolver.cs

```diff
@@ -215,12 +216,28 @@ orderby monikerGroup
         return model;
     }
 
-    private IReadOnlyDictionary<string, InternalXrefSpec[]> BuildInternalXrefMap()
+    public Dictionary<string, ExternalXrefSpec> ResolveXrefMapByFile(
+        FilePath file)
     {
-        var result = _internalXrefMapBuilder.Build();
-        ValidateUidGlobalUnique(result);
-        ValidateExternalXref(result);
-        return result;
+        var internalXrefSpecs = _internalXrefMap.Value.xrefsByFilePath.GetValueOrDefault(file, Array.Empty<InternalXrefSpec>());
+        var xrefMap = new Dictionary<string, ExternalXrefSpec>();
+        internalXrefSpecs.ToList().ForEach(spec =>
+        {
+            var externalXrefSpec = spec.ToExternalXrefSpec();
+            externalXrefSpec.SpecType = SpecType.Uid;
```

Comment:
The function `ToExternalXrefSpec` can take a parameter `SpecType`, so the  `SpecType` can be `init` only